src/kernels/miopen_rocrand.hpp

```diff
@@ -25,39 +25,51 @@
  *******************************************************************************/
 #pragma once
 
-namespace miopen::prng {
-// borrowed from rocrand, since hiprtc/comgr has some problems to compile kernels with rocrand
-struct xorwow_state
-{
-    // Weyl sequence value
-    unsigned int d;
-    // Xorshift values (160 bits)
-    unsigned int x[5];
-};
+#include "miopen_limits.hpp"
+#include "miopen_cstdint.hpp"
 
-inline constexpr unsigned int xorwow_next(xorwow_state* state)
-{
-    const unsigned int t = state->x[0] ^ (state->x[0] >> 2);
-    state->x[0]          = state->x[1];
-    state->x[1]          = state->x[2];
-    state->x[2]          = state->x[3];
-    state->x[3]          = state->x[4];
-    state->x[4]          = (state->x[4] ^ (state->x[4] << 4)) ^ (t ^ (t << 1));
+#ifndef WORKAROUND_IGNORE_ROCRAND_INCLUDES
```

Comment:
[R] If we have a ticket for rocrand, then it would be better to notice it somehow (WORKSROUND_SWDEV_XXX or just a comment). We can even open a ticket at https://github.com/ROCm/rocRAND -- that repo seems alive.